packages/prettier-plugin-java/src/printers/expressions.js

```diff
@@ -153,74 +154,63 @@ class ExpressionsPrettierVisitor {
   }
 
   binaryExpression(ctx) {
-    const sortedTokens = sortTokens(
-      ctx.Instanceof,
-      ctx.AssignmentOperator,
-      ctx.Less,
-      ctx.Greater,
-      ctx.BinaryOperator
-    );
     const referenceType = this.mapVisit(ctx.referenceType);
     const expression = this.mapVisit(ctx.expression);
     const unaryExpression = this.mapVisit(ctx.unaryExpression);
-    const segment = [unaryExpression.shift()];
-    for (let i = 0; i < sortedTokens.length; i++) {
-      const token = sortedTokens[i];
-      if (token.tokenType.tokenName === "Instanceof") {
-        segment.push(
-          rejectAndJoin(" ", [ctx.Instanceof[0], referenceType.shift()])
-        );
-      } else if (matchCategory(token, "'AssignmentOperator'")) {
-        segment.push(rejectAndJoin(" ", [token, expression.shift()]));
-      } else if (
-        i + 1 < sortedTokens.length &&
-        ((sortedTokens[i].image === ">" &&
-          sortedTokens[i + 1].image === ">" &&
-          sortedTokens[i].startOffset ===
-            sortedTokens[i + 1].startOffset - 1) ||
-          (sortedTokens[i].image === "<" &&
-            sortedTokens[i + 1].image === "<" &&
-            sortedTokens[i].startOffset ===
-              sortedTokens[i + 1].startOffset - 1))
-      ) {
-        if (
-          sortedTokens[i + 2] &&
-          sortedTokens[i + 2].image === ">" &&
-          sortedTokens[i + 1].startOffset ===
-            sortedTokens[i + 2].startOffset - 1
+
+    const { groups, sortedBinOps } = separateTokensIntoGroups(ctx);
+    const allSegment = [];
+    let tmpSegment = [];
+
```

Comment:
I would try to clarify what those variables mean. 

Maybe:
- groupsOfOperator instead of groups, 
- sortedBinaryOperators instead of sortedBinOps,
- segmentsSplittedByBinaryOperator instead of allSegments
- currentSegment instead of tmpSegment

WDYT?

Reply:
yep those variables fits it better :+1: